functests/performance/topology_manager.go

```diff
@@ -0,0 +1,194 @@
+package performance
+
+import (
+	"fmt"
+	"os/exec"
+	"path"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
+)
+
+var _ = Describe("[performance]Topology Manager", func() {
+	cs := testutils.NewClientSet("")
+	var workerRTNodes []corev1.Node
+
+	BeforeEach(func() {
+		var err error
+		workerRTNodes, err = nodes.GetByRole(cs, testutils.RoleWorkerRT)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(workerRTNodes).ToNot(BeEmpty())
+	})
+
+	It("should be enabled with the best-effort policy", func() {
+		kubeletConfig, err := nodes.GetKubeletConfig(cs, &workerRTNodes[0])
+		Expect(err).ToNot(HaveOccurred())
+
+		// verify topology manager feature gate
+		enabled, ok := kubeletConfig.FeatureGates[testutils.FeatureGateTopologyManager]
+		Expect(ok).To(BeTrue())
+		Expect(enabled).To(BeTrue())
+
+		// verify topology manager poicy
+		Expect(kubeletConfig.TopologyManagerPolicy).To(Equal(kubeletconfigv1beta1.BestEffortTopologyManagerPolicy))
+	})
+
+	Context("with the SR-IOV devices and static CPU's", func() {
+		var testpod *corev1.Pod
+		var sriovNode *corev1.Node
+
+		BeforeEach(func() {
+			sriovNodes := nodes.FilterByResource(cs, workerRTNodes, testutils.ResourceSRIOV)
+			if len(sriovNodes) < 1 {
+				Skip(
+					fmt.Sprintf(
+						"The environment does not have nodes with role %q and available %q resources",
+						testutils.RoleWorkerRT,
+						string(testutils.ResourceSRIOV),
+					),
+				)
+			}
+			sriovNode = &sriovNodes[0]
+
+			var err error
+			if testpod != nil {
```

Comment:
How could this happen?

Reply:
I assumed that we will have more SR-IOV topology manager tests under this context, so in general on the first test under the context `testpod` will be `nil`, but for all following tests it will be not `nil`

Reply:
Ah you are right :+1: